### ISSUE TITLE
Desktop icons become visible when changing workspace (21.10 beta)

### ISSUE THREAD

**Issue by hojjatabdollahi** [pop-os/shell#1244]
**(1) Issue/Bug Description:**
If you have multiple workspaces and on each one you have a window maximized, when switching between workspaces (no matter using the touchpad gesture or a keyboard shortcut) the desktop icons show up for a fraction of a second and then hide again behind the maximized windows.

**(2) Steps to reproduce (if you know):**
Have some icons on the desktop.
On a few workspaces, maximize a window.
Now switch between the workspaces.
The icons become visible while the animation for changing the workspace is happening.

**(3) Expected behavior:**
The icons should stay hidden behind the maximized windows.

**(4) Distribution (run `cat /etc/os-release`):**
 
Pop!_OS 21.10 Beta

**(5) Gnome Shell version:**
40.4.0

**(6) Pop Shell version (run `apt policy pop-shell` or provide the latest commit if building locally):**
```
pop-shell:
  Installed: 1.1.0~1636390326~21.10~b3595b2
  Candidate: 1.1.0~1636390326~21.10~b3595b2
  Version table:
 *** 1.1.0~1636390326~21.10~b3595b2 1001
       1001 http://apt.pop-os.org/release impish/main amd64 Packages
       1001 http://apt.pop-os.org/release impish/main i386 Packages
        100 /var/lib/dpkg/status
```

**(7) Where was Pop Shell installed from:**
Pop Repos.

**(8) Monitor Setup (2 x 1080p, 4K, Primary(Horizontal), Secondary(Vertical), etc):**
1x1080p

**(9) Other Installed/Enabled Extensions:**
Disabled all the other extensions except for Desktop Icon NG. 

**(10) Other Notes:**
Here is a video. 

https://user-images.githubusercontent.com/2222199/142259611-dcab8f52-7ed9-4265-a866-dbbd57911b43.mp4

**Comment by jacobgkau** [comment]
DING is showing up as manually-installed, even though it should be built-in. Did you install a different version from the built-in version intentionally?

Does the issue still occur if you disable the built-in Pop Shell extension?

**Comment by hojjatabdollahi** [comment]
Good point, I don't remember intentionally intalling DING, however, I might've visited extensions.gnome.org and noticed there is an update for DING and just pressed the update button.
I uninstalled the new DING and restarted the system, the builtin one appeared in the extension list. I disabled everything else except for cosmic workspaces and DING and the issue still exists. I did not have this issue before switching to the 21.10 beta, that's why I felt I should file a report.

https://user-images.githubusercontent.com/2222199/144097298-3d0046a2-4c81-45ae-8001-3313f5f6fa04.mp4

.
